internal/node/machine_mgr.go

```diff
@@ -162,61 +144,69 @@ func (m *MachineManager) DeployWorkload(vm *runningFirecracker, runRequest contr
+	} else if request.SupportsTriggerSubjects() {
+		for _, tsub := range request.TriggerSubjects {
 			_, err := m.nc.Subscribe(tsub, func(msg *nats.Msg) {
 				intmsg := nats.NewMsg(fmt.Sprintf("agentint.%s.trigger", vm.vmmID))
 				intmsg.Data = msg.Data
-				intmsg.Header.Add("x-nex-trigger-subject", msg.Subject)
+				intmsg.Header.Add(nexTriggerSubject, msg.Subject)
 
 				resp, err := m.ncInternal.RequestMsg(intmsg, time.Millisecond*10000) // FIXME-- make timeout configurable
 				if err != nil {
 					m.log.Error("Failed to request agent execution via internal trigger subject",
 						slog.Any("err", err),
 						slog.String("trigger_subject", tsub),
-						slog.String("workload_type", *runRequest.WorkloadType),
+						slog.String("workload_type", *request.WorkloadType),
 						slog.String("vmid", vm.vmmID),
 					)
-					functionFailedTriggers.Add(m.rootContext, 1)
-					functionFailedTriggers.Add(m.rootContext, 1, metric.WithAttributes(attribute.String("namespace", vm.namespace)))
-					functionFailedTriggers.Add(m.rootContext, 1, metric.WithAttributes(attribute.String("workload_name", *vm.deployedWorkload.WorkloadName)))
+
+					if m.t != nil {
+						m.t.functionFailedTriggers.Add(m.ctx, 1)
+						m.t.functionFailedTriggers.Add(m.ctx, 1, metric.WithAttributes(attribute.String("namespace", vm.namespace)))
+						m.t.functionFailedTriggers.Add(m.ctx, 1, metric.WithAttributes(attribute.String("workload_name", *vm.deployRequest.WorkloadName)))
+					}
 				} else if resp != nil {
-					runTime := resp.Header.Get("x-nex-run-nano-sec")
+					runtimeNs := resp.Header.Get(nexRuntimeNs)
 					m.log.Debug("Received response from execution via trigger subject",
 						slog.String("vmid", vm.vmmID),
 						slog.String("trigger_subject", tsub),
-						slog.String("workload_type", *runRequest.WorkloadType),
-						slog.String("function_run_time_nanosec", runTime),
+						slog.String("workload_type", *request.WorkloadType),
+						slog.String("function_run_time_nanosec", runtimeNs),
 						slog.Int("payload_size", len(resp.Data)),
 					)
-					runTime_int64, err := strconv.ParseInt(runTime, 10, 64)
+
+					runTimeNs64, err := strconv.ParseInt(runtimeNs, 10, 64)
 					if err != nil {
 						m.log.Warn("failed to log function runtime", slog.Any("err", err))
 					}
 
-					functionTriggers.Add(m.rootContext, 1)
-					functionTriggers.Add(m.rootContext, 1, metric.WithAttributes(attribute.String("namespace", vm.namespace)))
-					functionTriggers.Add(m.rootContext, 1, metric.WithAttributes(attribute.String("workload_name", *vm.deployedWorkload.WorkloadName)))
-					functionRunTimeNano.Add(m.rootContext, runTime_int64)
-					functionRunTimeNano.Add(m.rootContext, runTime_int64, metric.WithAttributes(attribute.String("namespace", vm.namespace)))
-					functionRunTimeNano.Add(m.rootContext, runTime_int64, metric.WithAttributes(attribute.String("workload_name", *vm.deployedWorkload.WorkloadName)))
+					if m.t != nil {
```

Comment:
I dont like this 😥
Lets initalize the counters always and make the provider conditional

Reply:
I agree, I did this temporarily to make everything work.